### PR TITLE
update filelist if a file has changed

### PR DESCRIPTION
**Pull request by newhinton**
## Description

When the Filesystem registers a change, only directory-changes trigger a sorting-action.
This PR adds that a filechange also triggers a sorting-action.

## Changes

Added else if-path for patchInFiletree for files

## Additional information

Tested on: Kubuntu 20.04

### CODE REVIEW

**Comment by newhinton** [comment]
See #2577

File: source/win-main/store.ts [hunk 1]
```diff
@@ -401,8 +401,12 @@ const config: StoreOptions<ZettlrState> = {
       // that the name did not change (because that would've resulted in a
       // removal and one addition) but rather something else, so we need to make
       // sure to simply re-sort it in case the sorting has changed.
+      // If we have a file, update the parent instead.
       if (ownDescriptor.type === 'directory') {
         Vue.set(ownDescriptor, 'children', sort(ownDescriptor.children, ownDescriptor.sorting))
+      } else if (ownDescriptor.type === 'file') {
+        const parentDescriptor = ownDescriptor.parent
```
Comment (nathanlesage) [inline source/win-main/store.ts:408]: `parent` is null if it's a standalone file, so this would throw an error.

**Review by nathanlesage — CHANGES_REQUESTED** [review]
Looks good to me except one comment!

File: source/win-main/store.ts (hunk 1 above)
Comment (newhinton) [reply source/win-main/store.ts]: i wrapped it in a nullcheck!

File: source/win-main/store.ts [hunk 2]
```diff
@@ -401,8 +401,14 @@ const config: StoreOptions<ZettlrState> = {
       // that the name did not change (because that would've resulted in a
       // removal and one addition) but rather something else, so we need to make
       // sure to simply re-sort it in case the sorting has changed.
+      // If we have a file, update the parent instead.
       if (ownDescriptor.type === 'directory') {
         Vue.set(ownDescriptor, 'children', sort(ownDescriptor.children, ownDescriptor.sorting))
+      } else if (ownDescriptor.type === 'file') {
+        if (ownDescriptor.parent != null) {
```
Comment (nathanlesage) [inline source/win-main/store.ts:408]: I would add that to the else if-condition (more style, but it makes the code a tad cleaner)

**Comment by nathanlesage** [comment]
One final comment, then it's ready to merge!

File: source/win-main/store.ts (hunk 2 above)
Comment (newhinton) [reply source/win-main/store.ts]: i kept the constant, i think it makes it more readable than reusing ownDescriptor.children.parent everywhere

Comment (nathanlesage) [reply source/win-main/store.ts]: I meant to add the `ownDescriptorparent != null` to the `else if (ownDescriptor.type === 'file')`-condition above.

**Comment by nathanlesage** [comment]
The Vue 3 migration has introduced merge conflicts, it would be super if you could fix them! Should work easily.

**Comment by newhinton** [comment]
I found the bug resulting in #2864. The fix is now in this PR aswell, should i make a second PR for this tiny change?

**Comment by nathanlesage** [comment]
> I found the bug resulting in #2864. The fix is now in this PR aswell, should i make a second PR for this tiny change?

Nah, we can add this in here as well. Just saw it, seems easy, understandable, and good. As long as it's in its own commit, all good!

**Review by nathanlesage — APPROVED** [review]

**Comment by nathanlesage** [comment]
And here we go!

**Pull request merged**